File: leetcode/100-999/125. Valid Palindrome/solution_2.cpp
class Solution
{
public:
    bool isPalindrome(string s)
    {
        int left = 0;
        int right = s.size() - 1;

        while (left < right)
        {
            auto leftCh = s[left];
            auto rightCh = s[right];

            if (!isalnum(leftCh))
                left++;

            if (!isalnum(rightCh))
                right--;

            if (isalnum(leftCh) && isalnum(rightCh))
            {
                if (tolower(leftCh) != tolower(rightCh))
                    return false;

                left++;
                right--;
            }

            return true;
        }
    };

File: leetcode/100-999/125. Valid Palindrome/solution_2.cs
public class Solution
{
    public bool IsPalindrome(string s)
    {
        int left = 0;
        int right = s.Length - 1;

        while (left < right)
        {
            var leftCh = s[left];
            var rightCh = s[right];

            if (!Char.IsLetterOrDigit(leftCh))
                left++;

            if (!Char.IsLetterOrDigit(rightCh))
                right--;

            if (Char.IsLetterOrDigit(leftCh) && Char.IsLetterOrDigit(rightCh))
            {
                if (Char.ToLower(leftCh) != char.ToLower(rightCh))
                    return false;

                left++;
                right--;
            }
        }

        return true;
    }
}